File: Euler7/Euler7.java
public class Euler7 {
     public static void main(String[] args){
          int primeCount = 1;
          long currentNum = 3L;
          while(primeCount < 10001){
            primeCount = (isPrime(currentNum)) ? primeCount + 1 : primeCount;
            currentNum += 2;
          }
          currentNum -= 2;
          System.out.println(currentNum);
     }

     private static boolean isPrime(long num){
          for (int i = 2; i*i <= num; i ++) {
              if (num % i == 0) return false;
          }
          return true;
     }
}

File: Python/euler7.py
import math

def euler7(n):
    count = 3
    focus = 5
    while(count < n+1):
        if (isPrime(focus)):
            count += 1
        focus += 2
    print(focus-2)

def isPrime(x):
    result = True
    myList = [i for i in range(3, math.ceil(math.sqrt(x)) + 1, 2)]
    
    for n in myList:
        if ((x % n) == 0):
            result = False
    return result


euler7(10001)